types/block.go

```diff
@@ -99,6 +105,11 @@ func (b Block) ID() BlockID {
 	return BlockID(crypto.HashObject(b.Header()))
 }
 
+// ID returns the ID of a Block, which is calculated by hashing the header.
+func (h BlockHeader) ID() BlockID {
```

Comment:
needs to be sorted by object. You've got a blockHeader function between a bunch of functions that operate on a block. I'd probably move it to above the block, though if you want to stick with the alphabetic convention you should move it below all the block functions.

Either one is fine with me, but I'd rather it not be in the middle of a bunch of block methods